### PR TITLE
raja: new port in devel

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

New port: https://github.com/LLNL/RAJA

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: devel/raja/Portfile [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# Dynamic lib disabled for a reason: https://github.com/LLNL/RAJA/issues/1431
+# Tests also fail to build with similar linking errors.
+# OneTBB not yet supported: https://github.com/LLNL/RAJA/issues/1456
+configure.args-append \
+                    -DCMAKE_INSTALL_PREFIX=${prefix} \
+                    -DENABLE_MPI=ON \
+                    -DENABLE_OPENMP=ON \
+                    -DRAJA_ENABLE_REPRODUCERS=ON \
+                    -DPython_EXECUTABLE=${prefix}/bin/python${py_ver} \
+                    -DBUILD_SHARED_LIBS=OFF \
```
Comment (reneeotten) [inline devel/raja/Portfile:60]: we typically prefer to build shared libs - why switch this off?

File: devel/raja/Portfile [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    system -W ${worksrcpath} "git submodule update --init"
+}
+
+compiler.cxx_standard   2014
+
+# Dynamic lib disabled for a reason: https://github.com/LLNL/RAJA/issues/1431
+# Tests also fail to build with similar linking errors.
+# OneTBB not yet supported: https://github.com/LLNL/RAJA/issues/1456
+configure.args-append \
+                    -DCMAKE_INSTALL_PREFIX=${prefix} \
```
Comment (reneeotten) [inline devel/raja/Portfile:55]: does the cmake PG not take care of this?

File: devel/raja/Portfile [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                    port:git
+    git.cmd         ${prefix}/bin/git
+}
+
+set py_ver          3.11
+set py_ver_nodot    [string map {. {}} ${py_ver}]
+depends_build-append \
+                    port:pkgconfig \
+                    port:python${py_ver_nodot}
+depends_lib-append  port:cppcheck
```
Comment (reneeotten) [inline devel/raja/Portfile:34]: is that really a build/runtime dependency?

File: devel/raja/Portfile [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+license             BSD
+maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
+description         RAJA Performance Portability Layer (C++)
+long_description    RAJA is a library of C++ software abstractions, primarily developed \
+                    at Lawrence Livermore National Laboratory (LLNL), that enables architecture \
+                    and programming model portability for HPC applications.
+
+fetch.type          git
+
+if {${os.major} < 14} {
```
Comment (reneeotten) [inline devel/raja/Portfile:21]: missing check for os.platform

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: devel/raja/Portfile (hunk 1 above)
Comment (barracuda156) [reply devel/raja/Portfile:60]: The reason is pointed out right above in a comment.

File: devel/raja/Portfile (hunk 2 above)
Comment (barracuda156) [reply devel/raja/Portfile:55]: I rebuilt it numerous times trying to get rid of Desul atomics, there was some weird behavior at destroot. But let me try removing and see if it works.

File: devel/raja/Portfile (hunk 3 above)
Comment (barracuda156) [reply devel/raja/Portfile:34]:
```
#------------------------------------
# Static analysis via Cppcheck
#------------------------------------
blt_find_executable(NAME        Cppcheck
                    EXECUTABLES cppcheck)
```
In `SetupThirdParty.cmake`. But I do not insist on it, and do not know if it actually does something.

Comment (barracuda156) [reply devel/raja/Portfile:34]: Yeah looks like it is not needed, I will remove it.

File: devel/raja/Portfile (hunk 2 above)
Comment (barracuda156) [reply devel/raja/Portfile]: You are right, thank you. My destroot troubles were caused by smth else. Removed.

**Review by reneeotten — APPROVED** [review]

**Comment by barracuda156** [comment]
@reneeotten Thank you!

**Pull request merged**